File: server/models/Captain.js
const mongoose = require('mongoose');
const bcrypt = require('bcryptjs');

const captainSchema = new mongoose.Schema({
    name: { type: String, required: true },
    email: { type: String, required: true, unique: true },
    password: { type: String, required: true },
    vehicle: {
        color: String,
        plate: String,
        capacity: Number,
        vehicleType: { type: String, enum: ['bike', 'auto', 'cab'] }
    },
    status: { type: String, enum: ['active', 'inactive'], default: 'inactive' },
    location: {
        lat: Number,
        lon: Number
    }
});

captainSchema.pre('save', async function (next) {
    if (!this.isModified('password')) return next();
    this.password = await bcrypt.hash(this.password, 10);
    next();
});

captainSchema.methods.comparePassword = async function (candidatePassword) {
    return await bcrypt.compare(candidatePassword, this.password);
};

module.exports = mongoose.model('Captain', captainSchema);

File: backend/src/main/java/com/swiftride/model/Captain.java
package com.swiftride.model;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
@Table(name = "captains")
public class Captain {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    
    @Column(unique = true)
    private String email;
    
    private String password;
    
    private String role = "captain";
    
    @Embedded
    private Vehicle vehicle;

    private String status = "inactive"; // active, inactive
    
    @Embedded
    private Location location;
}
